src/connect.rs

```diff
@@ -161,23 +161,29 @@ impl Connector {
             ProxyScheme::Http { .. } => {
                 unreachable!("connect_socks is only called for socks proxies");
             }
+            ProxyScheme::Https { .. } => {
+                unreachable!("connect_socks is only called for socks proxies");
+            },
         };
 
         match &self.inner {
             #[cfg(feature = "default-tls")]
             Inner::DefaultTls(_http, tls) => {
-                if dst.scheme() == "https" {
-                    use self::native_tls_async::TlsConnectorExt;
-
-                    let host = dst.host().to_owned();
-                    let socks_connecting = socks::connect(proxy, dst, dns);
-                    let (conn, connected) = socks::connect(proxy, dst, dns).await?;
+                if dst.scheme() == Some(&Scheme::HTTPS) {
+                    let host = dst.host().unwrap().to_owned();
+                    let conn = socks::connect(proxy, dst, dns).await?;
                     let tls_connector = tokio_tls::TlsConnector::from(tls.clone());
                     let io = tls_connector
                         .connect(&host, conn)
                         .await
                         .map_err(|e| io::Error::new(io::ErrorKind::Other, e))?;
-                    Ok((Box::new(io) as Conn, connected))
+                    Ok(Conn {
+                        inner: Box::new(NativeTlsConn { inner: io }),
+                        is_proxy: true,
+                    })
+                } else {
+                    // FIXME: should this be an error or just default to the third case?
+                    Err(Box::new(crate::error::builder("using connector for tls without https scheme?")))
                 }
```

Comment:
I'd appreciate if you could weigh in on this, @seanmonstar .

I added the `else` because the compiler (correctly) pointed out, that this `if` was missing an `else` branch (and was not returning `()`). 

However, I am not completely sure that it's correct to return an error in this case.

Reply:
You can see in the [0.9.x](https://github.com/seanmonstar/reqwest/blob/0.9.x/src/connect.rs#L162-L203) branch how this used to be. No `else` clause, instead it uses early `return`s, and falls-through to a default connect.

Reply:
ok fixed :)